internal/store.go

```diff
@@ -51,6 +51,7 @@ type Shard[K comparable, V any] struct {
 	vgroup    *Group[K, V] // used in secondary cache
 	counter   uint
 	mu        *RBMutex
+	closed    bool
```

Comment:
shouldn't this be an `atomic.Bool`? since you can have multiple goroutines trying to read & write to it?

Reply:
The `closed` field is protected by the shard mutex, and as you can see, I check it immediately before or after accessing the shard hashmap,  so the mutex is guaranteed to already be held at that point. Meanwhile, the `Close` method also first acquires the shard mutex before updating the `closed` field, making it safe to use a basic bool type here.

Reply:
I think it'd be good if you documented which fields are guarded by `mu`. For example: https://github.com/openfga/openfga/blob/94dd78f03dc99e51f46e7f71deaaea878e8340c9/pkg/storage/memory/memory.go#L125-L128

This doesn't change anything at runtime but it's useful documentation.